src/models/semantic.py

```diff
@@ -817,22 +836,30 @@ def on_test_epoch_end(self) -> None:
             self.test_cm.reset()
             return
 
+        if self.trainer.num_devices > 1:
+            epoch_cm = torch.sum(self.all_gather(self.test_cm.confmat), dim=0)
+        else:
+            epoch_cm = self.test_cm
+        # Reset metrics accumulated over the last epoch
+        self.test_cm.reset()
```

Comment:
I think the call to .reset() should happen last. You should compute and log your epoch metrics first and then log them. Maybe in DDP it will work because you copied the content of the metrics object from each process (to be verified) when calling all_gather() but I fear it won't work in the single-process case. Just to be sure, please move the .reset() calls to the end.